### ISSUE TITLE
Write a guide to making a good issue

### ISSUE THREAD

**Issue by Charlotteis** [yourfirstpr/yourfirstpr.github.io#2]
I would like to write a guide for project maintainers to encourage them to write good github issues.

**Comment by prologic** [comment]
Like any good project management a *good issue* should:

- have a short concise title that summarizes the issue
- have a description that **clearly** states what is to be solved

Often I find keeping new issues short and to the point helpful. Often discussions will arise from them and may either evolve the issue into something more concrete or may spawn a new issue that more clearly defines what is to be implemented or fixed.

I think for any project; too much "red tape" kills and dampens the project so keeping *initial issues* short and concise is **key**. Keeping discussions **open** and **constructive** help to evole issue(s) to maturity.

**Comment by Charlotteis** [comment]
I agree, @prologic. I think I would write a guide generally for writing good issues, as well as a guide to write issues specifically for junior developers and others who have never done Open Source before.

Thank you for your contribution! :clap:

**Comment by prologic** [comment]
Haha no worries! Just some thoughts I had!

On Monday, 7 September 2015, Charlotte Spencer <notifications@github.com>
wrote:

> I agree, @prologic <https://github.com/prologic>. I think I would write a
> guide generally for writing good issues, as well as a guide to write issues
> specifically for junior developers and others who have never done Open
> Source before.
>
> Thank you for your contribution! [image: :clap:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/yourfirstpr/yourfirstpr.github.io/issues/2#issuecomment-138291843>
> .
>


-- 

James Mills / prologic

E: prologic@shortcircuit.net.au
W: prologic.shortcircuit.net.au

**Comment by Charlotteis** [comment]
www.charlotteis.co.uk/making-your-first-pull-request/ starts to explain this.

**Comment by carlisia** [comment]
I'd like to make sure this resource is known so it can be included in the guide, it's super handy: https://github.com/blog/2111-issue-and-pull-request-templates

**Comment by mohammedri** [comment]
Hi there! I would like to give my 2 cents on issues:
Many times I see issues that folks post which may not seem as important to me vs. how excited they are about it. This always happens when they have a different context in their minds, then me/other developers, so the issue gets ignored or left behind. Therefore I think giving a good context (specially for feature requests) is a very important part of issues. 

_My preferred template, however is:_

_TL;DR: A good short summary._
For e.g. "Mozilla doesn't handle files served as */*+xml as XML files"; which is short, precise and gives the devs an idea of exactly what the problem and context is. (https://bugzilla.mozilla.org/show_bug.cgi?id=155730)
vs.
"Multiple reciepients are shown wrong"; where the context is slightly ambiguous.
(https://bugzilla.mozilla.org/show_bug.cgi?id=112785#c0)

_What is the problem?_
Ans. A detailed description of why this is a problem.

How did you notice/encounter this?
Ans. Context on how/what you were up-to when you encountered this.

**Steps to reproduce (if bug):**
1. Step 1
2. Step 2

_Expected results:_
...
_Actual results:_ 
...

Along with the above template, assigning appropriate labels, and informing the developers on any hunch you may have on how the problem could be fixed all together make a good issue imo.

If you are looking for a comprehensive study on what makes a good bug report, you can read this paper by Betternburg et al.
http://dl.acm.org/citation.cfm?id=1453146 

:)

**Comment by priyanshugandhi** [comment]
What is an issue?

Issue queues should mirror your thought process. Issues are where you structure your ideas, with others or alone.
Issues should focus on ideas, problems, and solutions; they are not activity trackers. While you want to share your ideas as they unfold, you may not want to comment every little bit of activity you register on an issue; for this, IM, commit messages or Scrum meeting may be better suited.
The issue queue is where real collaboration is done. Other forms of communication (verbal discussions, email, IM…) should happen after having explored things in an issue. This ensures that everyone involved had a chance to make and review the arguments in a clean, structured, and transparent manner. The more real-time the discussion, the more likely it is to be charged with emotions.
